x/lazyledgerapp/client/builder.go

```diff
@@ -35,8 +35,9 @@ func NewBuilder(address sdktypes.Address, chainID string) *Builder {
 	}
 }
 
-// UpdateAccountNumber queries the application to find the latest account number and
-// sequence, updating the respective internal fields
+// UpdateAccountNumber queries the application to find the latest account number
+// and sequence, updating the respective internal fields. This method should be
```

Comment:
btw minor nit: it might make sense to make the wording a bit stronger in case I understand this correctly: This method must be called instead of should.

Reply:
that makes more sense https://github.com/lazyledger/lazyledger-app/pull/94/commits/857ecb792b5ac8750d28d1ce4adc2dfefbe57273

I thought about including this method in `BuildSignedTx`, but I thought is was more expressive to the user as to what is actually going on by forcing them to call it. Do you think that makes sense? should I just go ahead and call it for them?

Reply:
> should I just go ahead and call it for them?

If there is no scenario the user will need control over this, then yeah it's probably a good idea to make this foolproof.

Reply:
per synchronous call, manual control of the account number is needed for "advanced" control over a given transaction, such as attempting to cancel an already broadcasted tx, raising the FeeAmount, or being clever with transaction ordering.